### ISSUE TITLE
rollbar:test task drops secret_key_base from config

### ISSUE THREAD

**Issue by jcoglan** [rollbar/rollbar-gem#964]
In our application, running `rake rollbar:test` fails because of the way that Rollbar clones the application with new config, in order to load its `/verify` endpoint. Specifically this section in `rollbar_test.rb` is broken on Rails 5.2 and possibly other versions:

https://github.com/rollbar/rollbar-gem/blob/v2.25.0/lib/rollbar/rollbar_test.rb#L92-L96

There are two issues with this code. First, it mutates the config of the original application object before making a new application, rather than cloning the config and changing the clone, which is not what the comments claim it does.

Second, it assumes that creating a new Rails app from the existing config object results in identical config except for the `cache_classes` change, which is not true. We put this in to check what was happening to `secret_key_base` in the cloned app:

```rb
  def self.rails_app
    # ...
    config = Rails.application.config
    config.cache_classes = true

    p [:before, config.object_id, config.secret_key_base]

    # Make a copy of the app, so the config can be updated.
    clone = Rails.application.class.name.constantize.new(:config => config)
    p [:after, clone.config.object_id, clone.config.secret_key_base]

    clone
  end
```

The `config` object on the cloned app is distinct from the original `config` object, and while `config.secret_key_base` has the value we set in our initialiser, `clone.config.secret_key_base` is `nil`. This prevents our app running successfully and results in the mock request sent by `rollbar:test` raising an exception as Rails cannot find the secret key base.

**Comment by waltjones** [comment]
@jcoglan When I test with a valid `Rails.application.credentials.secret_key_base`, the cloned app picks it up and the test works. How should I set up to reproduce what you're seeing?

**Comment by Menghongli** [comment]
I've tried to add host like this:
```
  def self.rails_app
    # ...
    config = Rails.application.config
    config.cache_classes = true
    config.hosts << 'www.example.com'

    # Make a copy of the app, so the config can be updated.
    Rails.application.class.name.constantize.new(:config => config)
  end
```

The generated app doesn't seems to pick up the right config.

**Comment by waltjones** [comment]
@jcoglan @Menghongli Wold love to get your feedback (if any) on this: https://github.com/rollbar/rollbar-gem/issues/979

**Comment by waltjones** [comment]
Fix released in v2.26.1 https://github.com/rollbar/rollbar-gem/releases/tag/v2.26.1

**Comment by jcoglan** [comment]
@waltjones I'm sorry I've not had the time to actually check that this works for us, but it looks like a sensible idea and removes all the code that was causing us problems. Thank you for fixing it :)
